lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -42,23 +84,41 @@ typedef _KeyDataResponseCallback = void Function(bool handled);
 class EnginePlatformDispatcher extends ui.PlatformDispatcher {
   /// Private constructor, since only dart:ui is supposed to create one of
   /// these.
-  EnginePlatformDispatcher._() {
+  EnginePlatformDispatcher() {
     _addBrightnessMediaQueryListener();
+    HighContrastSupport.instance.addListener(_updateHighContrast);
     _addFontSizeObserver();
+    registerHotRestartListener(dispose);
   }
 
   /// The [EnginePlatformDispatcher] singleton.
   static EnginePlatformDispatcher get instance => _instance;
-  static final EnginePlatformDispatcher _instance =
-      EnginePlatformDispatcher._();
+  static final EnginePlatformDispatcher _instance = EnginePlatformDispatcher();
 
   /// The current platform configuration.
   @override
   ui.PlatformConfiguration configuration = ui.PlatformConfiguration(
     locales: parseBrowserLanguages(),
     textScaleFactor: findBrowserTextScaleFactor(),
+    accessibilityFeatures: initializeAccessibilityFeatures(),
   );
 
+  /// Initialize accessibility features based on the current value of high contrast flag
+  static EngineAccessibilityFeatures initializeAccessibilityFeatures() {
```

Comment:
nit: typically `initialize` prefix implies mutation of some state. For example, there's a mutable static variable `foo` that starts as `null`, and then you call a `initializeFoo` function that makes that variable non-null. However, this function constructs the current state of `AccessibilityFeatures`. I'd therefore use a different prefix, such as `construct`, or `build`, or `compute`.